### DISCUSSION TITLE
参考文献里好多 .[S.l.]: [s.n.]

### DISCUSSION THREAD

**Discussion by Bennett561** [sjtug/SJTUThesis#743] [提问] [answered]
请问怎么把这个删掉呢？我查到其他学校模板里说可以在bst文件中改，但是咱们这里没找到bst文件……

**Comment by skyzh** [comment]
Hi, 是否有截图以及可供复现的文献 bibentry？

**Reply by Bennett561** [reply]
```
@inproceedings{uriarte2015system,
    title        = {System design and improvement of an emergency department using simulation-based multi-objective optimization},
    author       = {Uriarte, A Goienetxea and Z{\'u}{\~n}iga, E Ruiz and Moris, M Urenda and Ng, Amos HC},
    booktitle    = {Journal of physics: conference series},
    volume       = {616},
    number       = {1},
    pages        = {012015},
    year         = {2015},
    organization = {IOP Publishing}
}
```

**Comment by AlexaraWu** [comment, marked as answer]
修改 `setup.tex` 中的 `biblatex` 选项即可。

详见 Wiki [FAQ](https://github.com/sjtug/SJTUThesis/wiki/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98#%E4%B8%BA%E4%BB%80%E4%B9%88%E5%9C%A8%E5%8F%82%E8%80%83%E6%96%87%E7%8C%AE%E4%B8%AD%E4%BC%9A%E6%9C%89%E7%AC%A6%E5%8F%B7-279-279)。

**Reply by Bennett561** [reply]
好用，贴一下相关代码
`\usepackage[backend=biber,style=gb7714-2015,gbnamefmt=lowercase,gbpub=false]{biblatex}`

**Reply by zepinglee** [reply]
建议在模板中设置默认 `gbpub=false,`。

我原本也是设置默认按照国标的要求显示“出版地不详”或者“[S.l.]”等信息，同时也鼓励尽可能将出版信息填写完整。不过后来我在实际的使用过程中发现，有的文献真的很难找到出版地和出版社，尤其是会议论文（比如 ICLR, Neurips），会议举办地很容易找到，但是正式出版的出版社以及其地点查找起来却很麻烦。而且 `[S.l.]: [s.n.]` 是完全无意义的信息。

另外提议 @hushidong 在 `biblatex-gb7714-2015` 中将 `gbpub` 默认改为 `false`。

**Reply by Bennett561** [reply]
嗯嗯，确实

**Comment by note286** [comment]
@Bennett561 `tlmgr update --all --self` 更新一下包应该就可以解决了，TeX Live 2021内置的参考文献样式是2020发布的，是旧的，后面gb7714-2015包有更新。

**Reply by Bennett561** [reply]
谢谢，已经通过那个setup文件解决啦！
